src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -54,17 +68,62 @@ public override void Dispose()
     private static readonly byte[] _jsonComma = [Convert.ToByte(',')];
     private static readonly byte[] _jsonClosingBracket = [Convert.ToByte(']')];
 
-    public override async Task ProcessAsync(ArraySegment<byte> data)
+    public override async Task ProcessAsync(ArraySegment<byte> data, CancellationToken cancellationToken)
     {
         IncrementBytesReceivedMetric(data.Count);
-        PipeReader request = PipeReader.Create(new ReadOnlySequence<byte>(data.Array!, data.Offset, data.Count));
+
+        IMemoryOwner<byte> memoryOwner = MemoryPool<byte>.Shared.Rent(data.Count);
+        data.AsSpan().CopyTo(memoryOwner.Memory.Span);
+        Memory<byte> memory = memoryOwner.Memory[..data.Count];
+
+        await _processChannel.Writer.WriteAsync(new ProcessRequest(memory, memoryOwner), cancellationToken);
+    }
+
+    public override async Task ReceiveLoopAsync(CancellationToken cancellationToken)
+    {
+        using AutoCancelTokenSource cts = cancellationToken.CreateChildTokenSource();
+
+        using ArrayPoolList<Task> allTasks = new((int)(_processConcurrency + 1));
+        allTasks.Add(Task.Run(async () =>
```

Comment:
WDYT: Does it make sense to make this threaded/_long running task_, e.g.:
`Task.Factory.StartNew(f, cancellationToken, TaskCreationOptions.LongRunning);`

Reply:
Not sure. Probably.